src/api/product/services/getMany.js

```diff
@@ -6,9 +6,34 @@ async function readMany(req, res) {
   try {
     const limit = +req.query.limit;
     const pageNo = +req.query.pageNo;
+    const priceTo = +req.query.priceTo;
+    const priceFrom = +req.query.priceFrom;
+    const { language } = req.headers;
+    const { search } = req.query;
+    const { notSelectedLanguages } = req;
     const filter = { deleted: false };
 
-    return sendPaginatedList(res, Product, filter, pageNo, limit);
+    const dispatchedLanguage = {
+      [`name${req.headers.language}`]: 'name',
+      [`description${req.headers.language}`]: 'description',
+    };
+
+    const select = `-name${notSelectedLanguages[0]} -name${notSelectedLanguages[1]} -description${notSelectedLanguages[0]} -description${notSelectedLanguages[1]}`;
+    filter['$or'] = [
+      { [`name${language}`]: { $regex: `${search}`, $options: 'i' } },
+      { [`name${notSelectedLanguages[0]}`]: { $regex: `${search}`, $options: 'i' } },
+      { [`name${notSelectedLanguages[1]}`]: { $regex: `${search}`, $options: 'i' } },
+      { [`description${language}`]: { $regex: `${search}`, $options: 'i' } },
+      { [`description${notSelectedLanguages[0]}`]: { $regex: `${search}`, $options: 'i' } },
+      { [`description${notSelectedLanguages[1]}`]: { $regex: `${search}`, $options: 'i' } },
+    ];
+
+    if (priceTo) {
+      filter.price = { $lte: priceTo };
+    } else if (priceFrom) {
+      filter.price = { $gte: priceFrom };
+    }
```

Comment:
Lets speak about price filters.
In your code, if `priceTo` exists, your `priceFrom` will not be added, your `else if `will not be checked, because if your first condition is true in `if ()`, code will not check if `else if()` condition is true or false.
You have to check separately.
`if (priceTo) `- then attach your filter, then
`if (priceFrom)` - then attach your filter, and here remember that if both `priceTo` and `priceFrom` exist, your filter for price field must be different, both condition must be counted